litgpt/chat/base.py

```diff
@@ -110,6 +110,51 @@ def decode(fabric: L.Fabric, tokenizer: Tokenizer, token_stream: Iterator[torch.
     return tokens_generated
 
 
+def process_prompt(prompt, model, tokenizer, prompt_style, fabric, temperature, top_k, top_p, stop_tokens):
+    if not prompt.lower().strip() in ("", "quit", "exit"):
+        prompt = prompt_style.apply(prompt=prompt)
+        encoded_prompt = tokenizer.encode(prompt, device=fabric.device)
+        y = generate(
+            model, encoded_prompt, model.max_seq_length, temperature=temperature, top_k=top_k, top_p=top_p, stop_tokens=stop_tokens
+        )
+        fabric.print(">> Reply: ", end="")
+        t0 = time.perf_counter()
+        tokens_generated = decode(fabric, tokenizer, y)
+        t = time.perf_counter() - t0
+        for block in model.transformer.h:
+            block.attn.kv_cache.reset_parameters()
+        fabric.print(
+            f"\nTime for inference: {t:.02f} sec total, {tokens_generated / t:.02f} tokens/sec,"
+            f" {tokens_generated} tokens",
+            file=sys.stderr,
+        )
+        fabric.print()
+
+
+def interact(multiline_prompts, model, tokenizer, prompt_style, fabric, temperature, top_k, top_p, stop_tokens):
+    while True:
+        try:
+            if not multiline_prompts:
+                prompt = input(">> Prompt: ")
+            else:
+                print(">> Prompt: (Type '!submit' on a new line to end input)")
+                prompt_lines = []
+                while True:
+                    line = input()
+                    if line.strip().lower() == "!submit":
+                        break
+                    prompt_lines.append(line)
+                prompt = "\n".join(prompt_lines)
+
+        except KeyboardInterrupt:
+            break
+
+        if prompt.lower().strip() in ("quit", "exit"):
```

Comment:
Should we have !quit and !exit now?

Reply:
I think this makes sense for consistency. I updated it.